modules/system/console/OctoberUtil.php

```diff
@@ -294,7 +295,57 @@ protected function utilPurgeUploads()
             return;
         }
 
-        // @todo
+        $totalCount = 0;
+        $uploadsPath = Config::get('filesystems.disks.local.root', storage_path('app')) . '/uploads';
+
+        // Recursive function to scan the directory for files and ensure they exist in system_files.
+        $purgeFunc = function ($targetDir) use (&$purgeFunc, &$totalCount, $uploadsPath) {
+            if ($files = File::glob($targetDir.'/*')) {
+                if ($dirs = File::directories($targetDir)) {
+                    foreach ($dirs as $dir) {
+                        $purgeFunc($dir);
+
+                        if (File::isDirectoryEmpty($dir) && is_writeable($dir)) {
+                            rmdir($dir);
+                            $this->info('Removed folder: '. str_replace($uploadsPath, '', $dir));
+                        }
+                    }
+                }
+
+                foreach ($files as $file) {
+                    if (!is_file($file)) {
+                        continue;
+                    }
+
+                    if (!is_writeable($file)) {
+                        $this->warn('Unable to purge file: ' . str_replace($uploadsPath, '', $file));
+                        continue;
+                    }
+
+                    if (FileModel::where('disk_name', basename($file))->count()) {
+                        $this->warn('Skipped file in use: '. str_replace($uploadsPath, '', $file));
+                        continue;
+                    }
+
+                    // Skip .gitignore files
+                    if ($file === '.gitignore') {
```

Comment:
@bennothommo is it really needed to store `.gitignore` the file inside of upload folder? :)

Reply:
@w20k Yes, I believe the `storage/app/uploads` folder is stored in Git with a `.gitignore` file on a normal install - if it removes this file, people using Git might start seeing their uploads tracked in Git as well.

Reply:
See here: https://github.com/octobercms/october/tree/master/storage/app/uploads :)

Reply:
You're right @bennothommo that's a bit odd, though I do remember it was, sort of, an old school way of doing things.

GitIgnore files everywhere:
![Screenshot 2019-08-05 at 11 50 59](https://user-images.githubusercontent.com/1053320/62451779-9167c900-b777-11e9-8e5f-e4d9d2ded893.png)

Reply:
@w20k It's an old-school way, but it works - I imagine it ensures that the folders are there so people don't get any "missing folder" errors when uploading stuff. :)